### PR TITLE
feat: add Echarts gauge chart

### PR DESCRIPTION
**Pull request by krsnik93**
### SUMMARY
Add Echarts gauge chart
PR from superset-ui: https://github.com/apache-superset/superset-ui/pull/993

Screenshot:
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/visualizations/presets/MainPreset.js [hunk 1]
```diff
@@ -91,6 +92,7 @@ export default class MainPreset extends Preset {
         new EventFlowChartPlugin().configure({ key: 'event_flow' }),
         new FilterBoxChartPlugin().configure({ key: 'filter_box' }),
         new EchartsGraphChartPlugin().configure({ key: 'graph_chart' }),
+        new EchartsGaugeChartPlugin().configure({ key: 'gauge_chart' }),
```
Comment (etr2460) [inline superset-frontend/src/visualizations/presets/MainPreset.js:95]: looks like we're trying to keep this in abc order, so let's put this before the `graph_chart`

**Review by etr2460 — COMMENTED** [review]
could you also include a screenshot showing this chart selectable in explore in the test plan? thanks!

**Comment by krsnik93** [comment]
@etr2460 

Do you mean updating these two:
https://github.com/apache/superset/blob/master/superset-frontend/images/screenshots/visualizations.png
https://github.com/apache/superset/blob/master/superset-frontend/images/screenshots/visualizations.png

**Comment by etr2460** [comment]
@krsnik93 no need to update the visualizations.png file, i only meant to add a screenshot to the PR summary to verify it looks good. Although if we don't have a thumbnail in the viz selector yet, adding one there is a great idea too

**Review by etr2460 — APPROVED** [review]
this lgtm! thanks for the screenshot in the test plan

**Comment by junlincc** [comment]
A couple of cosmetic/overlapping issues in Customize tab. Not blocker, @krsnik93 can we open a follow up PR to address those after this PR is merged? Many thanks  🙏 @EBoisseauSierra @krsnik93 

1. 
Screen Shot 2021-04-13 at 9 51 00 AM

2.
Screen Shot 2021-04-13 at 9 52 18 AM

**Comment by krsnik93** [comment]
@junlincc Thanks for reporting this.

I can add more space between the axis and the label but then it won't look great for numbers with just a few digits. If you look closely at the second image, numbers like 0, 10, 20 etc look properly distanced from the axis. The issue is that Echarts seems to take the starting position and then render inwards, so if the number has a lot of digits, it ends up going over the axis. There is a single setting for label distance (We cannot set different values depending on the number of digits/chars etc.). I think the combination of tuning the split number (so that the values don't overlap between themselves) and the adaptive formatting (so that there are no values with a lot of digits) should be fine.

**Comment by krsnik93** [comment]
It seems like the overlap happens for all non-int labels so I will increase the label distance a bit.

**Comment by etr2460** [comment]
I'm going to merge this to get the new viz type launched (congrats!) and then we can follow up with further enhancements

File: superset-frontend/src/visualizations/presets/MainPreset.js (hunk 1 above)
Comment (ktmud) [reply superset-frontend/src/visualizations/presets/MainPreset.js:94]: This chart type should probably be renamed to "gauge".

**Comment by villebro** [comment]
@junlincc @krsnik93 I think we should try to get this fix in to ECharts, as I don't expect making that change should be very difficult. Let's open a ticket on the their GH Issues and see how the project feels about the proposed change of rendering the label outwards rather than inwards.

**Comment by krsnik93** [comment]
@villebro I had a look at the Echarts examples for gauge and I think they go inwards because the labels and ticks are originally placed on the inner side of the axis:

https://echarts.apache.org/examples/en/editor.html?c=gauge
https://echarts.apache.org/examples/en/editor.html?c=gauge-speed

When I was making the gauge plugin for Superset, I thought that having labels inwards would make them overlap more easily or overlap the value in the middle, so I applied a negative distance to have them on the outer side of the axis as here:

https://echarts.apache.org/examples/en/editor.html?c=gauge-temperature

So I think it's expected behavior from Echarts.

**Comment by villebro** [comment]
@krsnik93 ok now I understand 👍 In that case it could actually make sense to add an option to draw on the inside (default) or outside, with the assumption that outside will render outwards.

**Pull request merged**